distribution/lib/Standard/Base/0.0.0-dev/src/Error/File_Error.enso

```diff
@@ -13,30 +15,35 @@ type File_Error
 
        Arguments:
        - file: The file that doesn't exist.
-    Not_Found file
+    Not_Found (file : File)
 
     ## Indicates that a destination file already exists.
-    Already_Exists file
+    Already_Exists (file : File)
 
     ## A generic IO error.
 
        Arguments:
        - file: The file that couldn't be read.
        - message: The message for the error.
-    IO_Error file message
+    IO_Error (file : File) (message : Text)
 
     ## Indicates that the given file's type is not supported.
-    Unsupported_Type file
+    Unsupported_Type (file : File)
+
+    ## Indicates that the given file is corrupted, i.e. the data it contains
+       is not in the expected format.
+    Corrupted_Format (file : File) (message : Text) (cause : Any | Nothing = Nothing)
```

Comment:
Does Any not include Nothing? Just curious.

Reply:
It does, you are correct.

We sometimes use the `Any | Nothing` pattern to reinforce that a `Nothing` value is allowed here and it has some a bit special meaning.

How I'd read this signature is that the `cause` may or may not be present (`| Nothing`) and it may be of `Any` type (if it's present).

From typelevel perspective, just `Any` would suffice, but this is supposed to serve as kind of documentation.

I'm open to revising this approach though.

Reply:
No I think it makes sense to write it this way.